### PR TITLE
Remove config related to Sonar

### PR DESCRIPTION
**Pull request by samuelallan72**
We experimented with using Sonar,
but decided not to continue due to it not working as desired with our open source workflow.

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -35,13 +35,6 @@ jobs:
...
-    - name: Save PR number to file
-      run: echo ${{ github.event.number }} > PR_NUMBER.txt
-    - name: Archive PR number
-      uses: actions/upload-artifact@v4
-      with:
-        name: PR_NUMBER
-        path: PR_NUMBER.txt
     - name: Archive code coverage results
       uses: actions/upload-artifact@v4
       with:
```
Comment (chanchiwai-ray) [inline .github/workflows/check.yaml:40]: I believe this is also for sonar

**Review by chanchiwai-ray — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yaml (hunk 1 above)
Comment (samuelallan72) [reply .github/workflows/check.yaml]: +1

Comment (samuelallan72) [reply .github/workflows/check.yaml]: Updated :)

**Review by chanchiwai-ray — APPROVED** [review]
LGTM

**Review by rgildein — APPROVED** [review]

**Review by Pjack — APPROVED** [review]

**Pull request merged**